File: VB/WebUserControl.ascx.vb
Imports System
Imports DevExpress.Web.ASPxEditors
Imports DevExpress.Web.ASPxPopupControl

Partial Public Class WebUserControl
	Inherits System.Web.UI.UserControl

	Protected Sub ASPxPopupControl1_Init(ByVal sender As Object, ByVal e As EventArgs)
		Dim popupControl As ASPxPopupControl = DirectCast(sender, ASPxPopupControl)

		popupControl.ClientInstanceName = GetUserControlSpecificId("_popup")
		popupControl.HeaderText = Me.ID & "'s popup"
	End Sub

	Protected Sub btnShow_Init(ByVal sender As Object, ByVal e As EventArgs)
		Dim button As ASPxButton = DirectCast(sender, ASPxButton)

		System.Threading.Thread.Sleep(50)

		Dim randomizer As New Random()

		Dim clickEventCode As String = String.Format("function(s, e) {{    {0}.ShowAtPos({1}, {1});    }}", GetUserControlSpecificId("_popup"), 100 + randomizer.Next(Date.Now.Millisecond))

		button.ClientSideEvents.Click = clickEventCode
	End Sub

	Private Function GetUserControlSpecificId(ByVal elementId As String) As String
		Return Me.ID & elementId
	End Function
End Class

File: CS/WebUserControl.ascx.cs
using System;
using DevExpress.Web;

public partial class WebUserControl : System.Web.UI.UserControl {
    protected void ASPxPopupControl1_Init(object sender, EventArgs e) {
        ASPxPopupControl popupControl = (ASPxPopupControl)sender;
        popupControl.ClientInstanceName = GetUserControlSpecificId("_popup");
        popupControl.HeaderText = this.ID + "'s popup";
    }
    protected void btnShow_Init(object sender, EventArgs e) {
        ASPxButton button = (ASPxButton)sender;
        System.Threading.Thread.Sleep(50);
        Random randomizer = new Random();
        string clickEventCode = string.Format("function(s, e) { {0}.ShowAtPos({1}, {1}); }",
            GetUserControlSpecificId("_popup"), 100 + randomizer.Next(DateTime.Now.Millisecond));
        button.ClientSideEvents.Click = clickEventCode;
    }
    private string GetUserControlSpecificId(string elementId) {
        return this.ID + elementId;
    }
}
